### PR TITLE
Add test case for a pod becoming schedulable when a node is added

### PR DESCRIPTION
**Pull request by alculquicondor**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

We have little coverage around node addition and removal. Since distinct event handlers interact, it is important to cover this in integration tests. More cases can be added to exercise other events.

Part of #91111

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/scheduler/predicates_test.go [hunk 1]
```diff
@@ -1036,3 +1037,54 @@ var (
...
+			},
+			update: func(cs kubernetes.Interface) error {
+				_, err := createNode(cs, "node-1", nil)
+				return err
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (Huang-Wei) [inline test/integration/scheduler/predicates_test.go:1061]: I didn't see this pattern is used in integration test before. So I suppose it won't work as expected?

> // Parallel signals that this test is to be run in parallel with (and only with)
> // other parallel tests.

File: test/integration/scheduler/predicates_test.go [hunk 2]
```diff
@@ -1036,3 +1037,54 @@ var (
...
+			testCtx := initTest(t, "scheduler-informer")
+			defer testutils.CleanupTest(t, testCtx)
+			if tt.init != nil {
+				if err := tt.init(testCtx.ClientSet); err != nil {
+					t.Fatal(err)
+				}
+			}
+			tt.pod.Namespace = testCtx.NS.Name
+			pod, err := createPausePod(testCtx.ClientSet, initPausePod(testCtx.ClientSet, tt.pod))
```
Comment (Huang-Wei) [inline test/integration/scheduler/predicates_test.go:1070]: nit: cs is not used in `initPausePod `

**Review by Huang-Wei — COMMENTED** [review]
I think we need to add a couple of more subtests here.

**Review by alculquicondor — COMMENTED** [review]
Added a TODO for more test cases, which I will open to contributors.

File: test/integration/scheduler/predicates_test.go (hunk 1 above)
Comment (Huang-Wei) [reply test/integration/scheduler/predicates_test.go]: yup, creating nodes/pods with the same name won't make etcd happy.

**Comment by alculquicondor** [comment]
ping @Huang-Wei

**Pull request merged**